src/arc_utilities/ros_helpers.py

```diff
@@ -62,15 +62,18 @@ def logfatal(exception_class, msg):
     raise exception_class(msg)
 
 
-def get_connected_publisher(topic_path: str, *args, **kwargs):
+def get_connected_publisher(topic_path: str, *args, raise_on_failure: bool, **kwargs):
```

Comment:
Note, we should probably accept the message type here explicitly, rather than as part of `*args` or `**kwargs`

Reply:
ah yes good point!